### PR TITLE
Chore: refactor fastify.js

### PR DESCRIPTION
**Pull request by delvedor**
Hello team!

As all of you know, our entry point is a mess! 😱 
We worked really hard during the past year on adding new features and applied bugfixes, and we leave this file organization out in the cold.
This pr aims to start cleaning up this file. I can't do everything in this pr since it will be a massive change, furthermore, some work will be done in conjunction with https://github.com/fastify/fastify/issues/1503.

~~As you can see a test is failing, not sure how to fix it.~~

Related: https://github.com/fastify/fastify/issues/1475

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/internals/logger.test.js [hunk 1]
```diff
@@ -74,39 +72,6 @@ test('The logger should reuse request id header for req.id', t => {
...
-    })
-  })
-
-  fastify.inject({
-    method: 'GET',
-    url: '/'
-  }, function (err, res) {
-    t.error(err)
-  })
-})
```
Comment (mcollina) [inline test/internals/logger.test.js]: Why was this removed?

**Comment by mcollina** [comment]
I would also seek to refactor out the 404 handling, override and route definition, if possible.

File: test/internals/logger.test.js (hunk 1 above)
Comment (delvedor) [reply test/internals/logger.test.js]: We are already checking that in https://github.com/fastify/fastify/blob/834d3ee5817a5e16e6257898fd49aca241c94cf7/test/logger.test.js#L68

**Comment by delvedor** [comment]
> I would also seek to refactor out the 404 handling, override and route definition, if possible.

I agree, but I think we should split this work in multiple pull requests, otherwise, we will end up to break any other ongoing contribution.

File: lib/contentTypeParser.js [hunk 2]
```diff
@@ -216,13 +226,36 @@ function buildContentTypeParser (c) {
...
+  return this
+}
+
+function hasContentTypeParser (contentType) {
+  return this[kContentTypeParser].hasParser(contentType)
 }
 
 module.exports = ContentTypeParser
-module.exports.buildContentTypeParser = buildContentTypeParser
+module.exports.helpers = {
```
Comment (Ethan-Arrowood) [inline lib/contentTypeParser.js]: Are these helper functions exposed to the public API or just meant to be used internally? If they are going to be used externally I'd request we add some types; if not, no types are needed

Comment (delvedor) [reply lib/contentTypeParser.js]: The public API is the big `fastify` object you can see in the main faile :)

File: lib/logger.js [hunk 3]
```diff
@@ -72,6 +74,26 @@ function now () {
...
+  if (isValidLogger(options.logger)) {
+    const logger = createPinoLogger({
+      logger: options.logger,
+      serializers: Object.assign({}, serializers, options.logger.serializers)
+    })
+    return [logger, true]
+  } else if (!options.logger) {
+    const logger = Object.create(abstractLogging)
+    logger.child = () => logger
+    return [logger, false]
```
Comment (nwoltman) [inline lib/logger.js]: Using an object instead of an array would make this code easier to understand. With the array, it's not clear what `false` means here.

Comment (delvedor) [reply lib/logger.js]: Thank you for the feedback, but for now I'll leave it as is. By reading the code is easy to understand what that boolean is, and if this becomes a problem in the future I'll add some more doc.
Furthermore this syntax is becoming commonly used in JavaScript, so let's start embracing it :)

Comment (jsumners) [reply lib/logger.js]: I agree with @nwoltman. Just reading this function, I can't begin to tell you how this result will be used. Going over to `fastify.js` I can see that it is meant to be `{logger, hasLogger}`. I think this function should `return {logger, hasLogger: false}`.

See https://docs.google.com/document/d/1hWb-lQW4NSG9yRpyyiAA_9Ktytd5lypLnVLhPX9vamE/edit for information on array destructuring. I'm not opposed to array destructuring because of the doc, but it's something to keep in mind. I am simply opposed here due to legibility.
